fth/system.fth

```diff
@@ -596,6 +596,29 @@ ustack 0stackp
         char [compile] literal
 ; immediate
 
+: TOUPPER ( char -- char' )
+	dup [char] a >=
+	IF
+		dup [char] z <= IF [ char A char a - ] literal + THEN
+	THEN
+;
+
+: UPCASE ( c-addr u -- )
+	over + swap ?do
+		i c@ toupper i c!
+	loop
+;
+
+create WORD-SAVE-AREA 257 allot
```

Comment:
I think it would be OK to convert the output of parse-word at its original location. No need to copy it.